packages/integration/lib/src/widgets/post/comment/comment_reply_widget.dart

```diff
@@ -68,330 +63,228 @@ class _CommentReplyWidgetState extends State<LMFeedCommentReplyWidget> {
-              replies.insert(0, state.comment);
-            } else {
-              replies[index] = state.comment;
-            }
-          }
-        }
-        if (state is LMFeedEditLocalReplyState) {
-          int index =
-              replies.indexWhere((element) => element.id == state.replyId);
-          if (index != -1) {
-            LMCommentViewData reply = replies[index];
-            reply.text = state.text;
-            reply.isEdited = true;
-          }
-        }
-        if (state is LMFeedDeleteLocalReplyState) {
-          replies.removeWhere((element) => element.id == state.replyId);
-        }
-        if (state is LMFeedCommentRepliesLoadedState) {
-          Map<String, LMWidgetViewData>? widgets = state.commentDetails.widgets
-              ?.map((key, value) => MapEntry(
-                  key, LMWidgetViewDataConvertor.fromWidgetModel(value)));
-
-          Map<String, LMTopicViewData>? topics = state.commentDetails.topics
-              ?.map((key, value) =>
-                  MapEntry(key, LMTopicViewDataConvertor.fromTopic(value)));
-
-          users = state.commentDetails.users!.map((key, value) => MapEntry(
-              key,
-              LMUserViewDataConvertor.fromUser(value,
-                  topics: topics,
-                  userTopics: state.commentDetails.userTopics,
-                  widgets: widgets)));
-
-          users.putIfAbsent(user.uuid, () => user);
-          replies = state.commentDetails.postReplies!.replies
-                  ?.map((e) => LMCommentViewDataConvertor.fromComment(e, users))
-                  .toList() ??
-              [];
-        } else if (state is LMFeedPaginatedCommentRepliesLoadingState) {
-          Map<String, LMWidgetViewData>? widgets =
-              state.prevCommentDetails.widgets?.map((key, value) => MapEntry(
-                  key, LMWidgetViewDataConvertor.fromWidgetModel(value)));
-
-          Map<String, LMTopicViewData>? topics = state.prevCommentDetails.topics
-              ?.map((key, value) => MapEntry(key,
-                  LMTopicViewDataConvertor.fromTopic(value, widgets: widgets)));
-
-          users = state.prevCommentDetails.users!.map((key, value) => MapEntry(
-              key,
-              LMUserViewDataConvertor.fromUser(value,
-                  topics: topics,
-                  userTopics: state.prevCommentDetails.userTopics,
-                  widgets: widgets)));
+      listener: _handleListener,
+    );
+  }
 
-          users.putIfAbsent(user.uuid, () => user);
-          replies = state.prevCommentDetails.postReplies!.replies
-                  ?.map((e) => LMCommentViewDataConvertor.fromComment(e, users))
-                  .toList() ??
-              [];
+  void _handleListener(context, state) {
+    // handle comment state changes
+    // it will ignore the state changes of other comments
+    if (state is LMFeedGetReplyCommentSuccessState) {
+      if (state.commentId != widget.comment.id) {
```

Comment:
should use switch/case, and extract every sub check to another method for a specific case.

Reply:
switch case make make it more complex, i will try to extract this as much possible

Reply:
check for state only using switch, break down into mini handler functions

Reply:
done